lib/mongo/protocol/serializers.rb

```diff
@@ -50,7 +50,7 @@ module Header
         # @param value [ String ] Header value to be serialized.
         #
         # @return [ String ] Buffer with serialized value.
-        def self.serialize(buffer, value)
+        def self.serialize(buffer, value, validating_keys = nil)
```

Comment:
Not sure if these method signatures should be this instead:

```ruby
def self.serialize(buffer, value, validating_keys = BSON::Config.validating_keys?)
```
That would make a dependency on bson >= 4.1.0